### PR TITLE
New functions for adding date-of-interest reference lines

### LINKED ISSUE #53: Refactor dates to function
Since we might want to add arbitrary lines / dates to our graphs, it might be good to refactor and generalize the function for that:

https://github.com/LLNL/llnl.github.io/blob/0216a5d4a3b6914d5ac01f32e7d1d33bff88efb0/js/explore/line_repoActivityTotal.js#L122-L136

### PR DESCRIPTION
**Pull request by LRWeber**
Fixes #53

### CODE REVIEW

File: js/explore/line_repoActivityTotal.js [hunk 1]
```diff
@@ -44,6 +44,23 @@ function draw_line_repoActivityTotal(areaID) {
 		// Christmas
 		var dXmas = getYearDate("12-25",dToday);
 
+		function addDateLine(dateObj,label) {
```
Comment (IanLee1521) [inline js/explore/line_repoActivityTotal.js]: Would it be better to just take the date string itself as the argument? Is that `dateObj` used elsewhere?

**Comment by LRWeber** [comment]
The two are a bit different.

The creation history graph lines are of specific dates to-the-day, and include tool tips (though tricky to trigger) that show that exact date. This could potentially be dropped to make the functions identical.

The activity graph lines, meanwhile, have approximated dates with variable years in order to accommodate the constantly shifting timeline and shifting holidays (e.g. Thanksgiving being a "fourth Thursday" rather than a given date).

I could just use the date string as the argument for each function, though that would cause them to differ further: The first uses the complete date, while the second takes a month and day, then uses an additional "today" argument to calculate the appropriate year.
I could however solve that via a nested function, wrapping the necessary string conversion around a single shared date line function. Though I believe I would need to pass some additional arguments to the external function, like the chart itself.

Sorry for the brain dump, I probably shouldn't type while problem solving.
**TLDR**; I'll create the necessary solution to all of the above.

File: js/explore/helpers.js [hunk 2]
```diff
@@ -13,3 +13,21 @@ function GetTopX(arrayData, x) {
 	});
 	return arrayData.slice(0, x); 
 };
+
+// Draw date-of-interest reference lines on a graph
+function drawDateLine(dateObj,label,chart,x,y,height,valueline) {
```
Comment (IanLee1521) [inline js/explore/helpers.js]: We might want to add some comments / documentation at some point, but this looks good to me!
